pkg/cmd/infra/deployer/deployer.go

```diff
@@ -196,5 +203,53 @@ func (d *Deployer) Deploy(namespace, deploymentName string) error {
 	} else {
 		glog.Infof("Deploying from %s to %s (replicas: %d)", deployutil.LabelForDeployment(from), deployutil.LabelForDeployment(to), desiredReplicas)
 	}
-	return strategy.Deploy(from, to, desiredReplicas)
+	if err := strategy.Deploy(from, to, desiredReplicas); err != nil {
+		return err
+	}
+	return d.after(config, to)
```

Comment:
Why is this different from a post hook? Why should the deployment be marked as failed if our `AfterDeployment` actions fail?